src/quic/node_quic_util-inl.h

```diff
@@ -196,11 +196,21 @@ const uint8_t* QuicPreferredAddress::StatelessResetToken() const {
   return paddr_->stateless_reset_token;
 }
 
-const sockaddr_in6* QuicPreferredAddress::PreferredIPv6Address() const {
-  return reinterpret_cast<const sockaddr_in6*>(&paddr_->ipv6_addr);
+const std::string QuicPreferredAddress::PreferredIPv6Address() const {
+  char host[NI_MAXHOST];
+  // Return an empty string if unable to convert...
+  if (uv_inet_ntop(AF_INET6, paddr_->ipv6_addr, host, sizeof(host)) != 0)
+    return std::string();
+
+  return std::string(host);
 }
-const sockaddr_in* QuicPreferredAddress::PreferredIPv4Address() const {
-  return reinterpret_cast<const sockaddr_in*>(&paddr_->ipv4_addr);
+const std::string QuicPreferredAddress::PreferredIPv4Address() const {
```

Comment:
Adding `const` to return types only leads to compiler warnings :)

```suggestion
std::string QuicPreferredAddress::PreferredIPv4Address() const {
```